### PR TITLE
Feature: ORCID validator

### PR DESCRIPTION
**Pull request by andrewbrazzatti**
This pull request adds a new ORCID validator to the shared form validation logic and introduces comprehensive unit tests to ensure correct validation behavior for various ORCID input formats. The validator checks both the format and the checksum of ORCID identifiers, and the tests cover valid, invalid, and edge cases.

**Validation logic:**

- Added a new `orcid` validator to `formValidatorsSharedDefinitions` in `validators.ts`. This validator checks that the input matches the ORCID format (with or without hyphens), ensures the length is correct, and verifies the checksum digit using the official algorithm.

**Testing:**

- Added a suite of unit tests in `validators.test.ts` to cover the new ORCID validator. The tests verify acceptance of valid ORCID IDs (with and without hyphens, with "X" checksum), and rejection of various invalid formats, URLs, misplaced hyphens, incorrect lengths, invalid characters, and invalid checksums. Tests also confirm that empty and null values are accepted as valid (no error).

### CODE REVIEW

File: packages/sails-ng-common/src/validation/validators.ts [hunk 1]
```diff
@@ -374,4 +374,71 @@ export const formValidatorsSharedDefinitions: FormValidatorDefinition[] = [
...
+                actual: control.value,
+              },
+            },
+          };
+        }
+
+        const baseDigits = value.substring(0, 15);
+        const checkDigit = value.substring(15);
+
+        let total = 0;
```
Comment (cofiem) [inline packages/sails-ng-common/src/validation/validators.ts]: I suggest adding a comment with a link to the spec for the ORCID checksum algorithm.

File: packages/sails-ng-common/test/unit/validators.test.ts [hunk 2]
```diff
@@ -283,6 +283,163 @@ describe("Validator", async () => {
...
+                    args: {
+                        value: "0000-0002-1694-233X", definition: formValidatorsSharedDefinitions,
+                        block: {class: "orcid"},
+                    },
+                    expected: null,
+                },
+                {
+                    title: "orcid - expect failure (URL https)",
+                    args: {
+                        value: "https://orcid.org/0000-0002-1825-0097", definition: formValidatorsSharedDefinitions,
```
Comment (cofiem) [inline packages/sails-ng-common/test/unit/validators.test.ts]: Are urls ever valid ORCIDs? If not, then I think the 4 url tests are excessive - only one or two are needed.

Comment (andrewbrazzatti) [reply packages/sails-ng-common/test/unit/validators.test.ts]: Had them there because originally the code was allowing the full URL. The problem with that is that the format is inconsistent and makes using the ORCID more complicated. Now the code just allows formats that if you prepend `https://orcid.org/<number>`, the URL resolves.
